### PR TITLE
Make PCCMG Side-Specific Sidesets Generation Optional

### LINKED ISSUE #23724: Make Side-specific Sidesets Generated by PCCMG Optional
## Reason
PolygonConcentricCircleMG creates a series of side-specific sidesets with ids 1000x and 1500x. Also this is useful for some users who want to distinguish different sides of a polygon in their simulation. These extra sidesets are not needed in some applications, especially some Cardinal applications.

## Design
Make the generation of the 1000x and 1500x serieses sidesets optional (controlled by an input parameter)

## Impact
More user friendly.

### PR DESCRIPTION
**Pull request by miaoyinb**
closes #23724

### CODE REVIEW

**Comment by miaoyinb** [comment]
@GiudGiud This is a quick update to enable optional removal of some sidesets that are troublesome in Cardinal applications based on @aprilnovak's feedback. Thanks for your review in advance.

File: modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C [hunk 1]
```diff
@@ -160,6 +160,9 @@ PolygonConcentricCircleMeshGeneratorBase::validParams()
                         "Whether the outward interface boundaries are created.");
   params.addParam<boundary_id_type>(
       "interface_boundary_id_shift", 0, "Integer used to shift interface boundary IDs.");
+  params.addParam<bool>("generate_side_specific_boundaries",
+                        true,
```
Comment (GiudGiud) [inline modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C:164]: This is true for the assmbly?

**Review by GiudGiud — DISMISSED** [review]

File: modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C (hunk 1 above)
Comment (aprilnovak) [reply modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C:164]: Are there more users who want this setting enabled, or disabled? Would a default of `false` cover a wider user base?

Comment (GiudGiud) [reply modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C:164]: I d rather it be disabled by default, unless it s needed for building an assembly in the regular user workflow, in which case I would rather it be enabled

Comment (GiudGiud) [reply modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C:164]: not seeing too many failures with false as the default
```
test:meshgenerators/patterned_cartesian_mesh_generator.square_boundary ...................... FAILED (EXODIFF)
test:meshgenerators/patterned_hex_mesh_generator.hexagon_boundary ........................... FAILED (EXODIFF)
test:meshgenerators/patterned_cartesian_mesh_generator.square_boundary_no_interface ......... FAILED (EXODIFF)
test:meshgenerators/patterned_hex_mesh_generator.hexagon_boundary_no_interface .............. FAILED (EXODIFF)
test:meshgenerators/patterned_cartesian_peripheral_modifier.sq_pattern ...................... FAILED (EXODIFF)
test:meshgenerators/polygon_concentric_circle_mesh_generator.full ........................... FAILED (EXODIFF)
test:meshgenerators/polygon_concentric_circle_mesh_generator.full_no_interfaces ............. FAILED (EXODIFF)
test:meshgenerators/polygon_concentric_circle_mesh_generator.full_inward_interfaces ......... FAILED (EXODIFF)
test:meshgenerators/peripheral_ring_mesh_generator.err_invalid_boundary_type_2 FAILED (EXPECTED ERROR MISSING)
```

notably the RGMB workflow is fine 

As you prefer

Comment (aprilnovak) [reply modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C:164]: If the setting does cause failures sometimes, it'd probably be good to add an error message telling the user exactly how to fix their input files (you may already have this).

Comment (miaoyinb) [reply modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C:164]: Sure, I can set the default to false then, Do you prefer me regold these failed test or set the parameter to true in the inputs?

Comment (GiudGiud) [reply modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C:164]: no regolds if we can avoid it please

**Review by GiudGiud — APPROVED** [review]

**Comment by GiudGiud** [comment]
Thanks for the feedback @aprilnovak and thanks for the adjustment @miaoyinb !

**Pull request merged**